core/http/client/src/main/java/org/eclipse/rdf4j/http/client/SparqlSession.java

```diff
@@ -172,6 +188,24 @@ public SparqlSession(HttpClient client, ExecutorService executor) {
 		// parser used for processing server response data should be lenient
 		parserConfig.addNonFatalError(BasicParserSettings.VERIFY_DATATYPE_VALUES);
 		parserConfig.addNonFatalError(BasicParserSettings.VERIFY_LANGUAGE_TAGS);
+
+		// configure the maximum url length for SPARQL query GET requests
+		int maximumUrlLength = DEFAULT_MAXIMUM_URL_LENGTH;
+		String propertyValue = System.getProperty(MAXIMUM_URL_LENGTH_PARAM);
+		if (propertyValue != null) {
+			try {
+				maximumUrlLength = Integer.parseInt(propertyValue);
+			}
+			catch (NumberFormatException e) {
+				throw new RDF4JException("integer value expected for property " + MAXIMUM_URL_LENGTH_PARAM,
+						e)
+				{
+
+					private static final long serialVersionUID = 1L;
```

Comment:
We don't have a generic one. There's RepositoryConfigException and SailConfigException. 

We could introduce a generic RDF4JConfigException, and make the above two subclasses of that. It wouldn't break existing code if we did this AFAICT.